### PR TITLE
drop excess and duplicate logger properties

### PR DESCRIPTION
**Pull request by dprotaso**
Slack thread: https://knative.slack.com/archives/C964EFQAW/p1610738747008200

`zapdriver` package isn't maintained and has some weird quirks causing dupe info in our log output.

thus the following remains the same:
- all keys remain stable as before
- encoding levels are mapped to stack driver levels
- encoding time is RFC3339 nano

following changed (since we're not wrapping the core):
- I changed the duration output to a string that includes units (cc @mattmoor @vagababov)
- de-duped `caller` and sourceLocation
- no more empty `googleapis.com/labels` fields
- [this isn't invoked anymore](https://github.com/blendle/zapdriver/blob/9200777f8a3d4173247db5db22f6463772bb6f20/core.go#L112-L141) 
  - so no labels extraction to the above labels field (we weren't really use it)
  - no service context field attachment

/assign @yanweiguo @ian-mi

### CODE REVIEW

**Comment by dprotaso** [comment]
cc @slinkydeveloper given https://github.com/knative/pkg/pull/1773#issuecomment-745350926

**Comment by slinkydeveloper** [comment]
Sounds like a good change, thanks @dprotaso !

File: logging/config.go [hunk 1]
```diff
@@ -249,3 +251,9 @@ func ConfigToJSON(cfg *Config) (string, error) {
 	})
 	return string(jsonCfg), err
 }
+
+func stackdriverConfig() zap.Config {
```
Comment (yanweiguo) [inline logging/config.go:255]: Why is this called `stackdriverConfig`?

Comment (dprotaso) [reply logging/config.go:255]: `zapdriver` is specific to stack driver - I figured that'd be more explicit than `zapdriverConfig`

**Pull request merged**